@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -831,107 +845,156 @@ export class VisibleTileSet {
-                    continue;
-                }
+                const geoBox = this.getGeoBox(tilingScheme, childTileKey, offset);
+
+                if (useElevationRangeSource) {
+                    const range = elevationRangeSource!.getElevationRange(childTileKey);
+                    geoBox.southWest.altitude = range.minElevation;
+                    geoBox.northEast.altitude = range.maxElevation;
 
-                if (dataSource.shouldRender(displayZoomLevel, tileKey)) {
-                    visibleTiles.push(tileEntry);
+                    frustumIntersection.calculationFinal =
+                        frustumIntersection.calculationFinal &&
+                        range.calculationStatus === CalculationStatus.FinalPrecise;
                 }
 
-                tilingScheme.getSubTileKeys(tileKey).forEach(childTileKey => {
-                    const offset = tileEntry.offset;
-                    const tileKeyAndOffset = TileOffsetUtils.getKeyForTileKeyAndOffset(
-                        childTileKey,
-                        offset
-                    );
-                    const intersectsFrustum = tileFrustumIntersectionCache.get(tileKeyAndOffset);
+                let subTileArea = 0;
 
-                    if (intersectsFrustum !== undefined) {
-                        return;
+                if (this.projection.type === ProjectionType.Spherical) {
+                    const obb = new OrientedBox3();
+                    this.options.projection.projectBox(geoBox, obb);
+                    if (obb.intersects(this.m_frustum)) {
+                        subTileArea = 1;
                     }
+                } else {
+                    this.options.projection.projectBox(geoBox, tileBounds);
+                    subTileArea = this.computeSubTileArea(tileBounds);
+                }
 
-                    const geoBox = this.getGeoBox(tilingScheme, childTileKey, offset);
-
-                    if (useElevationRangeSource) {
-                        const range = elevationRangeSource!.getElevationRange(childTileKey);
-                        geoBox.southWest.altitude = range.minElevation;
-                        geoBox.northEast.altitude = range.maxElevation;
-
-                        allBoundingBoxesFinal =
-                            allBoundingBoxesFinal &&
-                            range.calculationStatus === CalculationStatus.FinalPrecise;
-                    }
+                if (subTileArea > 0) {
+                    const subTileEntry = new TileKeyEntry(childTileKey, subTileArea, offset);
+                    frustumIntersection.tileKeyEntries.set(tileKeyAndOffset, subTileEntry);
+                    workList.push(subTileEntry);
+                }
+            });
+        }
+        return frustumIntersection;
+    }
 
-                    let subTileArea = 0;
+    // Computes the visible tiles for each supplied datasource.
+    private getVisibleTilesForDataSources(
```

Comment:
Should be renamed, the visible tile keys are returned, and not the visible tiles.

Reply:
done